build-tools/automation/azure-pipelines.yml

```diff
@@ -0,0 +1,215 @@
+# Xamarin.Android Pipeline
+
+trigger:
+  - master
+  - d16-?
+
+# Global variables
+variables:
+  BundleArtifactName: bundle
+  AutoProvisionArgs: /p:AutoProvision=True /p:AutoProvisionUsesSudo=True /p:IgnoreMaxMonoVersion=False
+
+stages:
+- stage: prepare
+  displayName: Prepare
+  variables:
+    Configuration: Release
+    MSBuildAbiArgs: >-
+      /p:AndroidSupportedTargetJitAbis=armeabi-v7a:arm64-v8a:x86:x86_64
+      /p:AndroidSupportedHostJitAbis=Darwin:mxe-Win32:mxe-Win64
+      /p:AndroidSupportedTargetAotAbis=armeabi-v7a:arm64:x86:x86_64:win-armeabi-v7a:win-arm64:win-x86:win-x86_64
+  jobs:
+  - job: create_bundle
+    displayName: Create Bundle
+    pool: Hosted Mac Internal
+    timeoutInMinutes: 60
+    cancelTimeoutInMinutes: 5
+    steps:
+    - script: make prepare-deps V=1 CONFIGURATION=$(Configuration) MSBUILD_ARGS="$(AutoProvisionArgs) $(MSBuildAbiArgs)"
+      displayName: make prepare-deps
+
+    - task: MSBuild@1
+      displayName: create bundle
+      inputs:
+        solution: build-tools/create-bundle/create-bundle.csproj
+        configuration: $(Configuration)
+        msbuildArguments: /bl:bin/Build$(Configuration)/msbuild-createbundle.binlog $(MSBuildAbiArgs)
+
+    - task: CopyFiles@2
+      displayName: copy bundle
+      inputs:
+        sourceFolder: bin/$(Configuration)
+        contents: bundle-*.zip
+        targetFolder: $(Build.ArtifactStagingDirectory)
+
+    - task: PublishPipelineArtifact@0
+      displayName: upload bundle
+      inputs:
+        artifactName: $(BundleArtifactName)
+        targetPath: $(Build.ArtifactStagingDirectory)
+
+    - task: MSBuild@1
+      displayName: package build results
+      inputs:
+        solution: build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks.csproj
+        configuration: $(Configuration)
+        msbuildArguments: /t:ZipBuildStatus /p:ResultPackageOutputPath=$(Build.ArtifactStagingDirectory)/results
+      condition: always()
+
+    - task: PublishPipelineArtifact@0
+      displayName: upload artifacts
+      inputs:
+        artifactName: prepare-build-results
+        targetPath: $(Build.ArtifactStagingDirectory)/results
+      condition: always()
+
+- stage: mac_build
+  displayName: Mac
+  dependsOn: prepare
+  jobs:
+  - job: mac_build
+    displayName: Build
+    pool: Hosted Mac Internal
+    timeoutInMinutes: 240
+    cancelTimeoutInMinutes: 5
+    variables:
+      Configuration: Release
+      JavaSdkDirectory: /Library/Java/JavaVirtualMachines/jdk1.8.0_144.jdk/Contents/Home/
```

Comment:
Is this and the following line strictly necessary?  The provisionator script (should?) ensure that *only* JDK 1.8.0_144 is present *anyway*, and if we *don't* specify it here, then it means there's one less location to update when the JDK *is* changed...

Reply:
Since `JAVA_HOME` points to something else on the hosted mac machines, we'll want to set that variable for now. This can be cleaned up in the future when we change our JDK provisioning approach.